fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/cascades/expressions/RelationalExpression.java

```diff
@@ -121,9 +121,9 @@ public interface RelationalExpression extends Correlated<RelationalExpression>,
     static RelationalExpression fromRecordQuery(@Nonnull RecordMetaData recordMetaData,
                                                 @Nonnull RecordQuery query) {
         query.validate(recordMetaData);
-        final var allRecordTypes = recordMetaData.getRecordTypes().keySet();
-        final var recordTypesFromQuery = query.getRecordTypes();
-        final var queriedRecordTypes = recordTypesFromQuery.isEmpty() ? allRecordTypes : recordTypesFromQuery;
+        final Set<String> allRecordTypes = recordMetaData.getRecordTypes().keySet();
```

Comment:
you don't like `var`s?

Reply:
In general, no, not really. In this case, it's not _just_ persnickety-ness, but it was legitimately hard to tell here whether it was using `RecordType`s or `String`s, which made it hard to understand